### DISCUSSION TITLE
Tasks being retried even though retries = 0

### DISCUSSION THREAD

**Discussion by cesar-vermeulen** [apache/airflow#33612] [General]
Hello everybody,

Since a couple of weeks, we have been struggling with a hard situation that did not occur in the past (~3 weeks).
Some context:

We run Airflow on AKS (2.6.2 in prod and 2.7 in acpt - issue occurring in both environments), and during the night, the bulk of the load is being triggered.
We have a task in nearly each DAG at a specific time frame, start_azure_data_factory (in the future i'll call this start adf), with afterwards a sensor wait_for_data_factory.

Now, the retries on the start adf are explicitly set to 0:
![image](https://github.com/apache/airflow/assets/94971679/154aff69-e3af-4882-99c6-70c67f8459a7)

Which is reflected in the task configuration:
![image](https://github.com/apache/airflow/assets/94971679/8e4fef52-9e6d-45b4-b92b-1d210790268c)

However, we notice that for some DAG runs, Airflow decides on retrying this task after X amount of minutes (ranging between 5 to 10 minutes). This causes duplicates in our data as the same table is being ingested twice. 

Some extra information:
We see that Airflow is attempting to adopt the pod, but suddenly decides on creating a second attempt : 
![image](https://github.com/apache/airflow/assets/94971679/8065f178-13ef-41af-88fd-85135be45ad0)
![image](https://github.com/apache/airflow/assets/94971679/d4b26eca-57aa-48fd-80fa-365e82024a42)

It is possible that our AKS cluster is experiencing resource problems and/or pods stay in pending status for > 5 minutes - could this have to do something with the retry_delay?

Some additional deployment details coming from earlier discussions/questions:
- Running 3 schedulers
- Running on KubernetesExecutor
- Postgres as a backend, but no resource constraints there:
![image](https://github.com/apache/airflow/assets/94971679/86aaaf3a-8f57-497e-85cf-fba3a6903b59)
- orphaned_tasks_check_interval: 300.0 
- scheduler_zombie_task_threshold: 900
- Task typically runs for ~ 1 - 5 secs

Is there anyone who could point me in the right direction to start looking? This is quite a nasty problem and there is no real root-cause to be determined. Happy to provide more details if necessary.

Thanks in advance!

**Comment by cesar-vermeulen** [comment]
Still struggling with this issue, if anyone could take a look at this, would be awsome!

**Comment by potiuk** [comment]
I think you should create an issue from it and provide a few examples of logs - Ideally wiht more logging context and specifically with log lines visible in the log.
